packages/webapp/src/components/Animals/SexDetails/index.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { useMemo, useState } from 'react';
+import SexDetailsPopover, { Details } from './SexDetailsPopover';
+import InputBase from '../../Form/InputBase';
+import styles from './styles.module.scss';
+import { ReactComponent as ChevronDown } from '../../../assets/images/chevron-down-2.svg';
+import { useTranslation } from 'react-i18next';
+
+type SexDetailsProps = {
+  initialDetails: Details;
+  maxCount: number;
+  onConfirm: (d: Details) => void;
+  onCancel?: () => void;
+};
+
+const initialize = (details: Details) => () => structuredClone(details) as Details;
+
+export default function SexDetails({
+  maxCount,
+  initialDetails,
+  onCancel,
+  onConfirm,
+}: SexDetailsProps) {
+  const [anchor, setAnchor] = useState<HTMLElement | null>(null);
+  const [details, setDetails] = useState(initialize(initialDetails));
+  const total = details.reduce((prevCount, { count }) => prevCount + count, 0);
+  const unspecified = maxCount - total;
+  const isPopoverOpen = !!anchor;
+  const { t } = useTranslation();
+
+  const handleCancel = () => {
+    setAnchor(null);
+    setDetails(initialize(initialDetails));
+    onCancel?.();
+  };
+
+  const handleConfirm = (details: Details) => {
+    setAnchor(null);
+    onConfirm(details);
+  };
+
+  const handleCountChange = (id: Details[0]['id'], count: number) =>
+    setDetails(
+      details.map((detail) => {
+        if (detail.id === id) detail.count = count;
+        return detail;
+      }),
+    );
+
+  // memoizing ensures this readonly input does not update while user is editing the popover
+  const anchorInput = useMemo(
+    () => (
+      <InputBase
+        label={t('ADD_ANIMAL.SEX_DETAIL')}
+        optional
+        error={total > maxCount ? t('ADD_ANIMAL.SEX_DETAIL_ERROR', { count: maxCount }) : undefined}
+        showResetIcon={false}
+        rightSection={<ChevronDown className={styles.chevronDownIcon} />}
+        mainSection={
```

Comment:
I used a button instead of the default input here because in the designs the input display is "Male: {count} - Female: {count}" where count is in bold and there is no way to style input values like that as far as I know. Also a button is more accessible because you can open the popover/modal by tabbing to it and pressing enter or space.

Reply:
I think it's a good idea! You could probably use `<TextButton />`.

Reply:
I didn't realize we had a text button, thanks for the suggestion!